addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -698,6 +698,33 @@ export class OdooEditor extends EventTarget {
                 this.updateColorpickerLabels();
             });
         }
+        // Handle the font size input.
+        const fontSizeInput = this.toolbar.querySelector('input#fontSizeCurrentValue');
+        this.addDomListener(this.toolbar, 'click', ev => {
+            if (fontSizeInput && ev.target.closest('#font-size .dropdown-toggle')) {
+                this._lastMousedown = undefined;
+                // If the click opened the font size dropdown, select the input content.
+                fontSizeInput.select();
+            } else if (!this.isSelectionInEditable()) {
+                // Otherwise, if we lost the selection in the editable, restore it.
+                this.historyResetLatestComputedSelection(true);
+            }
+        })
+        if (fontSizeInput) {
+            this.addDomListener(this.document, 'mousedown', ev => (this._lastMousedown = ev.target));
+            this.addDomListener(fontSizeInput.ownerDocument, 'mousedown', ev => (this._lastMousedown = ev.target));
+            this.addDomListener(fontSizeInput, 'change', () => {
+                // Don't apply if the user clicked on one of the dropdown items.
```

Comment:
Is this really a use case ? Wouldn't the change event be called after the click one ? I suppose it's best not to rely on that but the trick here to avoid that is a little convoluted anyway IMO... Another option I have in mind could be that the dropdown buttons actually don't change the font size in the dom but change the value of the input then manually trigger a change, that way the only code that actually sets the font size would be the input change.

Reply:
I believe it's a real use case: you click on the dropdown, start typing, then change your mind and just click one of the dropdown items. Without this, it will apply the value of the input (because apparently the change event is called first). It's true the trick is a bit convoluted, that's the best method I could find... Your suggestion is not bad, I could do that, but that's not as small a change as it seems because the dropdown works with the xml `data-call="setFontSize"` construction. So that would mean removing that and manually binding each of them. But it's maybe best to do it anyway indeed...

Reply:
Well even without rebinding them manually we could have `data-call="setFontSizeInput"` instead, couldn't we ?

Reply:
Hmmm I guess we could, it's a bit of a weird command but sure. And it would trigger a change event then?

Reply:
Wait, it's an editor _command_ ? I thought it was like a function name on some js class. Clearly we're not having an editor command for that so probably setting it manually would be easier. It's not like the rest of the widget isn't already handled in js anyway.

Reply:
That's right. And indeed it's easier to do it manually...